File: board-back/src/main/java/com/example/boardback/dto/request/auth/SignUpRequestDto.java
package com.example.boardback.dto.request.auth;

import javax.validation.constraints.AssertTrue;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class SignUpRequestDto {
  
  @NotBlank @Email// null이 아니면서 빈문자열이 아니고 공백이 아님
  private String email;
  @NotBlank @Size(min=8, max=20)
  private String password;
  @NotBlank
  private String nickname;
  @NotBlank @Pattern(regexp="^[0-9]{11,13}$")
  private String telNumber;
  @NotBlank
  private String address;

  private String addressDetail;
  @NotNull @AssertTrue // 참조형 변수에 사용할수있음
  private Boolean agreedPersonal;
}

File: board-back-nest/src/modules/auth/dto/request/sign-up.request.dto.ts
import { Equals, IsBoolean, IsEmail, IsNotEmpty, IsOptional, Matches, MaxLength, MinLength } from "class-validator";

export default class SignUpRequestDto {
  @IsNotEmpty()  // NotBlack 같음
  @IsEmail()
  email: string;

  @IsNotEmpty()
  @MinLength(8)
  @MaxLength(20)
  password: string;

  @IsNotEmpty()
  nickname: string;

  @IsNotEmpty()
  @Matches(/^[0-9]{11,13}$/) // 정규식
  telNumber: string;
  
  @IsNotEmpty()
  address : string;

  @IsOptional()  // 있으면 검증하고 없으면 검증하지마라
  addressDetail: string | null;

  @IsBoolean()
  @Equals(true) // 어떤 것과 같은지 체크하겠다 -> 이 경우는 true
  agreedPersonal : boolean;
}